### PR TITLE
Support deployment MaxSurge

### PR DESCRIPTION
**Pull request by hangyan**
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kube-state-metrics/365)

### CODE REVIEW

File: collectors/deployment.go [hunk 1]
```diff
@@ -83,6 +83,12 @@ var (
 		[]string{"namespace", "deployment"}, nil,
 	)
 
+	descDeploymentStrategyRollingUpdateMaxSurge = prometheus.NewDesc(
```
Comment (andyxning) [inline collectors/deployment.go]: @hangyan You should also add `ch <- descDeploymentStrategyRollingUpdateMaxSurge` to `Describe` function to tell prometheus client go scrape the max surge metrics.

Others lgtm

**Comment by andyxning** [comment]
@hangyan Add a comment. PTAL.

File: collectors/deployment_test.go [hunk 2]
```diff
@@ -40,6 +40,9 @@ var (
 
 	depl1MaxUnavailable = intstr.FromInt(10)
 	depl2MaxUnavailable = intstr.FromString("20%")
+
+	depl1MaxSurge = intstr.FromInt(10)
+	depl2MaxSurge = intstr.FromString("20%")
```
[the remark this answers is not in the document]
Comment (hangyan) [reply collectors/deployment_test.go]: @andyxning Updated,thanks! also add test for `depl2MaxUnavailable `

**Pull request merged**